lib/highLevelConsumer.js

```diff
@@ -171,45 +170,38 @@ HighLevelConsumer.prototype.connect = function () {
                             return p.topic;
                         });
                         self.client.refreshMetadata(topicNames, function (err) {
-                            register();
                             self.rebalancing = false;
                             if (err) {
-                              self.emit('error', err);
+                                self.emit('error', err);
                             } else {
-                              self.emit('rebalanced');
+                                self.emit('rebalanced');
+                            }
+                            if (self.pending) {
+                                self.pending = false;
+                                rebalance();
                             }
                         });
                     }
                 });
             });
+        } else {
+            self.pending = true;
         }
     }
 
-    // Wait for the consumer to be ready
-    this.on('registered', function () {
```

Comment:
`consumersChanged` will trigger whenever `init` method completes, so there is no necessary to listen to `registered` event.